core/src/blocks/browser.rs

```diff
@@ -149,27 +149,47 @@ impl Block for Browser {
             },
         }?;
 
+        let query = r#"
+mutation ScrapeWebsite($url: String!, $timeout: Float, $selector: String = "body") {
+    goto(url: $url, waitUntil: firstContentfulPaint, timeout: $timeout) {
+        status
+        time
+        url
+        status
+    }
+    body: html(selector: $selector) {
+        html
+    }
+    text: text(selector: $selector) {
+        text
+    }
+}
+"#;
+
         let mut body_json = json!({
-            "url": url,
-            "elements": [ { "selector": self.selector } ],
-            "gotoOptions": {
-                "timeout": self.timeout,
-            },
+            "query": query,
+            "variables": {
+                "url": url,
+                "selector": self.selector,
+                "timeout": self.timeout
+            }
         });
 
+        // WARN: not sure what to do with it. There is no waitFor in the current /scrape route, could be
+        // waitForTimeout or waitForNavigation (or else?)
```

Comment:
Looking at the existing calls, we only have 2 apps which define "waitFor" - one with a selector, another with a timeout .. 
`"waitFor": "section[aria-label='Menu']"`
`"waitFor": 10000 `

I don't think it's anyway doing anything .. ?

Reply:
On the previous API i don't know, couldn't find any documentation on that. But for the new one, we could switch based on string/int, on the API there is a `waitForTimeout` and `waitForSelector`.

Reply:
(the 2 apps are actually in the same ws : https://dust.tt/poke/cf9d79bf9d , looks like tests )